pkg/lib/dpop/middleware.go

```diff
@@ -47,17 +48,18 @@ func (m *Middleware) Handle(next http.Handler) http.Handler {
 }
 
 func (m *Middleware) handleError(rw http.ResponseWriter, err error) {
-	if apierrors.IsAPIError(err) {
-		apierr := apierrors.AsAPIError(err)
+	var oauthErr *protocol.OAuthProtocolError
+	if errors.As(err, &oauthErr) {
 		rw.Header().Set("Content-Type", "application/json")
 		rw.Header().Set("Cache-Control", "no-store")
 		rw.Header().Set("Pragma", "no-cache")
-		errJson := map[string]string{
-			"error":             "invalid_token",
-			"error_description": apierr.Message,
-		}
+		errJson := oauthErr.Response
 		errJsonStr, _ := json.Marshal(errJson)
-		http.Error(rw, string(errJsonStr), apierr.Name.HTTPStatus())
+		statusCode := oauthErr.StatusCode
+		if statusCode == 0 {
+			statusCode = http.StatusBadRequest
+		}
+		http.Error(rw, string(errJsonStr), statusCode)
```

Comment:
You cannot use http.Error https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/net/http/server.go;l=2180 It writes plain text response.

Instead, you just use

```
rw.WriteHeader(statusCode)
rw.Write([]byte(errJsonStr))
```